homeassistant/components/hvv_departures/sensor.py

```diff
@@ -0,0 +1,214 @@
+                )
+
+                self.attr[ATTR_LINE] = departure["line"]["name"]
+                self.attr[ATTR_ORIGIN] = departure["line"]["origin"]
+                self.attr[ATTR_DIRECTION] = departure["line"]["direction"]
+                self.attr[ATTR_TYPE] = departure["line"]["type"]["shortInfo"]
+                self.attr[ATTR_ID] = departure["line"]["id"]
+                self.attr[ATTR_DELAY] = departure.get("delay", 0)
+
+                departures = []
+                for departure in data["departures"]:
+                    departures.append(
+                        {
+                            ATTR_DEPARTURE: departure_time
+                            + timedelta(minutes=departure["timeOffset"])
+                            + timedelta(seconds=departure.get("delay", 0)),
+                            ATTR_LINE: departure["line"]["name"],
+                            ATTR_ORIGIN: departure["line"]["origin"],
+                            ATTR_DIRECTION: departure["line"]["direction"],
+                            ATTR_TYPE: departure["line"]["type"]["shortInfo"],
+                            ATTR_ID: departure["line"]["id"],
+                            ATTR_DELAY: departure.get("delay", 0),
+                        }
+                    )
+                self.attr[ATTR_NEXT] = departures
+            else:
+                self._available = False
+
+        except InvalidAuth as error:
+            if self._last_error != InvalidAuth:
+                _LOGGER.error("Authentication failed: %r", error)
+                self._last_error = InvalidAuth
+            self._available = False
+        except ClientConnectorError as error:
+            if self._last_error != ClientConnectorError:
+                _LOGGER.warning("Network unavailable: %r", error)
+                self._last_error = ClientConnectorError
+            self._available = False
+        except Exception as error:  # pylint: disable=broad-except
+            if self._last_error != error:
+                _LOGGER.error("Error occurred while fetching data: %r", error)
+                self._last_error = error
+            self._available = False
+
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this sensor."""
+        station_id = self.config_entry.data["station"]["id"]
+        station_type = self.config_entry.data["station"]["type"]
+
+        return f"{DOMAIN}-{self.config_entry.entry_id}-{station_id}-{station_type}"
+
+    @property
+    def device_info(self):
+        """Return the device info for this sensor."""
+        return {
+            "identifiers": {
+                (
+                    DOMAIN,
+                    self.config_entry.entry_id,
+                    self.config_entry.data["station"]["id"],
+                    self.config_entry.data["station"]["type"],
+                )
+            },
+            "name": self.config_entry.data["station"]["name"],
```

Comment:
```suggestion
            "name": self._name,
```

Reply:
Actually, "Departures at Wartenau" is not the right device name. It should be the station name as before, as there will be a binary_sensor platform for the elevator status, and then the "Departures at.. " in the device name would be wrong

Reply:
I re-thought the whole naming thing. My integration is called "HVV Departures" and the main thing about this integration is the departures sensor. The upcoming elevator status is just supplementary information for the departure. So I'm gonna leave the device name like you suggested.